uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,447 @@ define([
     templateUrl: ['settings', function (settings) {
       return settings.pathTpl + 'components/manager-leave-requests.html';
     }],
-    controllerAs: 'requests',
-    controller: ['$log', controller]
+    controllerAs: 'ctrl',
+    controller: ['$log', '$q', 'Contact', 'AbsencePeriod', 'AbsenceType', 'LeaveRequest',
+      'OptionGroup', controller]
   });
 
 
-  function controller($log) {
+  function controller($log, $q, Contact, AbsencePeriod, AbsenceType, LeaveRequest, OptionGroup) {
     $log.debug('Component: manager-leave-requests');
+
+    var vm = Object.create(this),
+      allStatus = {
+        name: 'all',
+        label: 'All'
+      };
+
+    vm.isFilterExpanded = false;
+    vm.absencePeriods = [];
+    vm.filteredUsers = [];
+    vm.absenceTypes = [];
+    //leaveRequests.table - to handle table data
+    //leaveRequests.filter - to handle left nav filter data
+    vm.leaveRequests = {
```

Comment:
Changed to but alphabetical order, but did not create a single object as I think that will make less readable.